\documentclass[handout,xcolor=pdftex,dvipsnames,table]{beamer} 

\usecolortheme[RGB={0,0,144}]{structure}
%\usetheme{AnnArbor}\usecolortheme{beaver}
\usetheme{CambridgeUS}\usecolortheme{dolphin}

\usepackage{graphicx}
\usepackage[space]{grffile}
\usepackage{verbatim,xmpmulti,color,multicol,multirow}
\setlength{\unitlength}{\textwidth}  % measure in textwidths
\usepackage[normalem]{ulem}
\usepackage{amssymb,amsmath,latexsym}
\usepackage{booktabs}
\usepackage{array}
\newcolumntype{L}{>{$}l<{$}}
\newcolumntype{C}{>{$}c<{$}}
\newcolumntype{R}{>{$}r<{$}}
\newcommand{\nm}[1]{\textnormal{#1}}


%\usepackage{beamerthemesplit}
\setbeamertemplate{navigation symbols}{}
\setbeamertemplate{caption}[numbered]
%\setbeamercolor{alerted text}{fg=red}
%\setbeamertemplate{block body theorem}{bg=orange}
\setkeys{Gin}{width=0.6\textwidth}


%\SweaveOpts{concordance=TRUE}

<<setup, include=FALSE>>=
#render_listings()
#pdf.options(useDingbats = TRUE)
opts_chunk$set(fig.path='figure/beamer-',fig.align='center',fig.show='hold',size='footnotesize',echo=FALSE,message=FALSE,error=FALSE,warning=FALSE)
opts_chunk$set(fig.width=5, fig.height=5, out.width='.6\\linewidth', fig.align='center')

@

<<echo=FALSE,results='hide'>>=
# some setup
options(width=60) # make the printing fit on the page
set.seed(1121) # make the results repeatable
require(xtable)
require(PASWR)
@

\title[RNASeq Analysis]{Analyze RNASeq Data from G9P2 RFI (Residual Feed Intake) Lines Using QuasiSeq Package}
\date{August 27, 2014}

% \author[Yet Nguyen, Dan Nettleton]{Yet Nguyen \and Dan Nettleton}
% \institute[ISU]{Iowa State University}

%

\begin{document}
\frame{\maketitle}
\frame {
\frametitle{Table of Contents}
\begin{itemize}
\setlength{\itemsep}{.2in}
\item[1.] Data Summary
\item[2.] Model Selection Strategies
\begin{itemize}
\item Number of genes with pvalues less than 0.05
\item Distance between Grenander CDF Estimator to Uniform CDF using
\begin{itemize}
\item Anderson-Darling Statistics
\item Cram{\'e}r-Von-Mises statistics
\item Kolmogorov-Smirnov statistics 
\end{itemize}
\end{itemize}
\item[3.] Results
\begin{itemize}
\item Generation 9 Data (single end and paired end data)
\item Generation 8 Data
\item Comparison between 2 Generations
\end{itemize}
\end{itemize}
}


\section{Data Summary}
\begin{frame}[fragile]
\frametitle{RNASeq Data}
\begin{itemize}
\setlength{\itemsep}{.25in}
\item RNASeq data set is a  25320 $\times$ 31 table of count data  corresponding to 25320 genes of 31  pigs from 2 Lines: high RFI Line and low RFI Line, and 2 Diets: high energy diet (Diet 1) and low energy diet (Diet 2).

\item  For Diet 1, the RNA data are from 7 low RFI line pigs and 8  high RFI line pigs. 
For Diet 2, the RNA data are from 8 low RFI line pigs and 8  high RFI line pigs.
 
\end{itemize}
 

\end{frame}




%%%%%%%%%%%%%%%%
\frame{
\frametitle{Covariates}
\begin{itemize}
\setlength{\itemsep}{.25in}
\item The available metadata consists of infomation of  9 covariates for 31 samples of 31 pigs.

\begin{itemize}
\item Factors: Diet (2 levels), Line (2 levels), Block (4 levels), Blockorder (8 levels).
\item Quantitative covariates: RFI (RFI values) , RINb (RNA Integrity Number before globin depletion), RINa (RNA Integrity Number after globin depletion), Concb (RNA Concentration before globin depletion), Conca(RNA Concentration after globin depletion).
\end{itemize}
\item CBC (Complete Blood Count) data:  neutrophils, lymphocytes, monocytes, eosinophils, and basophils. The CBC covariates are in model in the form of log2 transformation.
\end{itemize}
}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%5

\frame{
\frametitle{Number of Genes Used in Analysis}
\begin{itemize}

\setlength{\itemsep}{.25in}


\item Single End Read Count Data: The number of genes analyzed is 12222. Those are genes with average counts greater than 8 and for which there are at least four samples with non-zero counts. 


\item Paired End Read Count Data: The number of genes analyzed is 12280. Those are genes with average counts greater than 8 and for which there are at least four samples with non-zero counts. 
\end{itemize}
}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section{Model Selection}
\begin{frame}
\frametitle{Model Selection Criteria}
Starting model includes all  covariates of interest. For each covariate, 
\begin{itemize}
\setlength{\itemsep}{.2in}
\item We conduct a Likelihood Ratio Test using QuasiSeq of the full model vs. the reduced model obtained from the full model by deleting the considered covariate. We collect the set of pvalues of all genes from the tests. 

\item Obtain the number of genes with pvalues less than or equal 0.05. 
\item Obtain Grenander CDF estimator of the empirical CDF of the sample from those pvalues.
\item Obtain the Anderson-Darling statistics, Cram{\'e}r-Von-Mises statistics, and Kolmogorov-Smirnov statistics between the Grenander CDF and uniform(0,1) CDF.
\end{itemize}
Exclude the covariate corresponding to the smallest value for most of the above criteria.
\end{frame}


\begin{frame}[fragile]
\frametitle{Grenander's Estimator}
<<results='hide',out.width='6cm', out.height='6cm', fig.show='hold'>>=
library("fdrtool")



# samples from random exponential variable 
set.seed(1)
z = rbeta(100, 1,4)
e = ecdf(z)
g = grenander(e)

# plot ecdf, concave cdf, and Grenander density estimator (on log2 scale)

x <- g
#dev.off()
hist(z,  prob = T,col = 2, main = 
       "Simulation Data z ~ dbeta(100,1,4)")

curve(dbeta(x, 1, 4), col = 3, add = TRUE, lwd = 2)
lines(x = c(0,0.8), y = c(1,1), col = 1, lwd = 2)
points(x$x.knots, x$f.knots, type="s",
  col=4, lwd=2)
legend("topright", legend = c("Grenander PDF", "dbeta(1,4)", "dunif(0,1)"), 
      col = c(4, 3, 1), lwd = 2)

  plot(x$F, do.points=FALSE, col = 2, pch = 1, lwd = 2, xlim = c(0, 1),
       ylim = c(0,1), main = "CDFs")
  lines(x$x.knots, x$F.knots, type='l', col=4, lwd=2)
  lines(x = c(0, 1), y = c(0, 1), type ="l", col = 1, lwd = 2)
curve(pbeta(x,1,4), col = 3, lwd = 2, add = TRUE)
legend("bottomright",legend = c("Empirical CDF","Grenander CDF",  "beta(1,4) CDF", 
                                "Uniform(0,1) CDF"),
   col = c(2,4,3, 1), lwd = 2)

  
# for comparison the kernel density estimate


@
\end{frame}
\section{Results}

\begin{frame}[fragile]
\frametitle{Single End Read Count Data-Backward Model Selection}
\begin{figure}[h!]
    \centering
    %\includegraphics[width=\textwidth,height=0.8\textheight,keepaspectratio]{/home/ntyet/research/RFI-newdata/analysis/rfi/Plot06_26_2014_1.pdf}
    %\includegraphics[width=\textwidth,height=1.4\textheight,keepaspectratio]{Plot07_022014_rficbc.pdf}
    
      \includegraphics[width=\textwidth,height=1.4\textheight,keepaspectratio]{Plot07_022014_rficbcsingleend.png}
    \end{figure}
\end{frame}



\begin{frame}[fragile]

\frametitle{Paired End Read Count Data-Backward Model Selection}
\begin{figure}[h!]
    \centering
    %\includegraphics[width=\textwidth,height=0.8\textheight,keepaspectratio]{/home/ntyet/research/RFI-newdata/analysis/rfi/Plot06_26_2014_1.pdf}
   % \includegraphics[width=\textwidth,height=1.4\textheight,keepaspectratio]{Plot07_022014_rficbcdat2.pdf}
     \includegraphics[width=\textwidth,height=1.4\textheight,keepaspectratio]{Plot07_022014_rficbcpairedend.png}
    \end{figure}
\end{frame}



\begin{frame}[fragile]
\frametitle{Number of Common DE Genes between two Data Sets}


<<>>=
#load("Model7_fit.RData")
load("Model7_result.RData")

@

<<results ='asis'>>=
### Reading data #######
covset <- read.csv("covset.csv")
attach(covset)
load("Model7_fit.RData")
# str(fit)
# dim(fit$coef)
# full_model <- model.matrix(~Line + Concb + RINa + 
#                              lneut + llymp + lmono + lbaso + 
#                              as.factor(Block))
#colnames(full_model)
load("Model7_result.RData")
scount <- read.table("single end uniquely mapped reads count table for Yet.txt", 
                     header = T)

counts <- as.matrix(scount[rowSums(scount[,-1]>0)>3&
                           rowMeans(scount[,-1])>8& 
                             rowSums(scount[,-1][,Line ==1] > 0) >0 &
                             rowSums(scount[,-1][, Line ==2] >0) >0 ,-1])
gene_name <- scount[rowSums(scount[,-1]>0)>3&
                           rowMeans(scount[,-1])>8& 
                             rowSums(scount[,-1][,Line ==1] > 0) >0 &
                             rowSums(scount[,-1][, Line ==2] >0) >0 ,][,1]

write.table(gene_name, "gene_list_g9_s.txt", 
            row.names = FALSE, 
            col.names = FALSE)
## List of Line DE Genes when FDR is controled at 0.05, 0.10, 0.15

degene05 <- which(result$Q.values[[3]][,"Line"]<=0.05)
degene10 <- which(result$Q.values[[3]][,"Line"]<=0.10)
degene15 <- which(result$Q.values[[3]][,"Line"]<=0.15)
degene20 <- which(result$Q.values[[3]][,"Line"]<=0.20)

gene_name_de05_s <- gene_name[degene05]

## Total number of Line  DE Genes when FDR is controlled at 0.05, 0.10, 0.15

degene <- c(length(degene05), length(degene10), length(degene15), length(degene20))

## List of subset of Line DE Genes with log2(FC) >=1 when FDR is controlled at 0.05, 0.10, 0.15

lf105 <- which((abs(fit$coef[degene05]))*log2(exp(1)) >=1)
gene_name_lf105_s <- gene_name[lf105]


#scount[as.numeric(names(lf105)),1]
lf110 <-which((abs(fit$coef[degene10]))*log2(exp(1)) >=1)

lf115 <- which((abs(fit$coef[degene15]))*log2(exp(1)) >=1)


lf120 <- which((abs(fit$coef[degene20])) >=1)
## Total number of Line DE Genes with log2(FC) >=1 when FDR is controlled at 0.05, 0.10, 0.15

lf1 <- c(length(lf105), length(lf110), length(lf115), length(lf120))

# Summary table
out1 <- data.frame(FDR =c(0.05, 0.10, 0.15,.20), degene = degene, lf1 = lf1)
colnames(out1) <- c("FDR", "DEGs", "$\\mbox{log2(FC)}\\geq 1$")
@


<<results='hide'>>=

## Covariate Set
covset <- read.csv("covset.csv")
attach(covset)

## Load the result of model 11
load("Model7_resultdat2.RData") 

## To get P.values of all Line Testings, use result$P.values[[3]][,"Line]
## To get Q.values of Line Testings, use result$Q.values[[3]][,"Line]

scount <- read.table("paired end uniquely mapped reads count table.txt", 
                     header = T)

## List of Genes used to find DE Genes
scount <- scount[-c(which(scount[,1] %in%"ENSSSCG00000007978"),
                    which(scount[,1] %in%"ENSSSCG00000014725")),]
counts <- as.matrix(scount[rowSums(scount[,-1]>0)>3&
                           rowMeans(scount[,-1])>8 ,-1])
#dim(scount)
gene_name_p <- scount[rowSums(scount[,-1]>0)>3&
                           rowMeans(scount[,-1])>8 ,][,1]
#length(gene_name_p)
write.table(gene_name_p, "gene_list_g9_p.txt",
            row.names = FALSE, 
            col.names = FALSE)
#scount[,1]
## List of Line DE Genes when FDR is controled at 0.05, 0.10, 0.15

degene05 <- which(result$Q.values[[3]][,"Line"]<=0.05)
head(scount[rownames(counts[degene05,]), 1]) # First 6 of the List of Line DE Genes when FDR is controled at 0.05
gene_name_de05_p <- gene_name_p[degene05]


degene10 <- which(result$Q.values[[3]][,"Line"]<=0.10)
degene15 <- which(result$Q.values[[3]][,"Line"]<=0.15)
degene20 <- which(result$Q.values[[3]][,"Line"]<=0.20)

## Total number of Line  DE Genes when FDR is controlled at 0.05, 0.10, 0.15

degene <- c(length(degene05), length(degene10), length(degene15),length(degene20))

## List of Line DE Genes with log2(FC) >=1 when FDR is controlled at 0.05, 0.10, 0.15

lf105 <- which((abs(fit$coef[degene05]))*log2(exp(1)) >=1)
gene_name_lf105_p <- gene_name[lf105]


#scount[as.numeric(names(lf105)),1]
lf110 <-which((abs(fit$coef[degene10]))*log2(exp(1)) >=1)

lf115 <- which((abs(fit$coef[degene15]))*log2(exp(1)) >=1)


lf120 <- which((abs(fit$coef[degene20]))*log2(exp(1)) >=1)

## Total number of Line DE Genes with log2(FC) >=1 when FDR is controlled at 0.05, 0.10, 0.15

lf1 <- c(length(lf105), length(lf110), length(lf115), length(lf120))
@



<<results='asis'>>=
# Summary table
out2 <- data.frame(FDR =c(0.05, 0.10, 0.15,.20), degene = degene, lf1 = lf1)
colnames(out2) <- c("FDR", "DEGs", "$\\mbox{log2(FC)}\\geq 1$")

#print(xtable(out2), include.rownames = F, sanitize.colnames.function = identity)
@

\begin{table}[htb]
\begin{minipage}{.45\textwidth}
\centering
<<model7singleendresult,echo=FALSE,results='asis'>>=

print(xtable(out1), include.rownames = F, sanitize.colnames.function = identity,
  floating=FALSE,
  hline.after=NULL,
  add.to.row=list(pos=list(-1,0, nrow(out1)),
  command=c('\\toprule\n','\\midrule\n','\\bottomrule\n')))
@
\captionof{Single End Data: }{Estimated number of DE Genes between 2 Lines is 4580.}
\end{minipage}
\begin{minipage}{.45\textwidth}
\centering
<<model8result,echo=FALSE,results='asis'>>=


print(xtable(out2), include.rownames = F, sanitize.colnames.function = identity,
  floating=FALSE,
  hline.after=NULL,
  add.to.row=list(pos=list(-1,0, nrow(out2)),
  command=c('\\toprule\n','\\midrule\n','\\bottomrule\n')))
@
\captionof{Paired End Data: }{Estimated number of DE Genes between 2 Lines is 4990.}
\end{minipage}
\end{table}


When FDR is controlled at $5\%$ in both dataset, there are 444 common DE genes between two RFI lines in which  441 common DE gene with log2fc at least 1.
% <<result ='hide'>>=
% length(intersect(gene_name_de05_s, gene_name_de05_p))
% length(intersect(gene_name_lf105_s, gene_name_lf105_p))
% @

\end{frame}

\begin{frame}[fragile]
\frametitle{Pvalues of Line Effects}

\begin{figure}
\centering
\begin{subfigure}
  \centering
  \includegraphics[width=.4\linewidth]{PvalueLineSingleEnd.png}
\end{subfigure}%
\begin{subfigure}
  \centering
  \includegraphics[width=.4\linewidth]{PvalueLinePairedEnd.png}
\end{subfigure}
\end{figure}
\end{frame}

\begin{frame}[fragile]
\frametitle{G8 - Backward  Model Selection I }
\begin{figure}[h!]
    \centering
    %\includegraphics[width=\textwidth,height=0.8\textheight,keepaspectratio]{/home/ntyet/research/RFI-newdata/analysis/rfi/Plot06_26_2014_1.pdf}
    \includegraphics[width=\textwidth,height=1.4\textheight,keepaspectratio]{PlotmodelcbcdateRNA.png}
    \end{figure}
\end{frame}

\begin{frame}[fragile]
\frametitle{G8 - Backward  Model Selection II }
\begin{figure}[h!]
    \centering
    %\includegraphics[width=\textwidth,height=0.8\textheight,keepaspectratio]{/home/ntyet/research/RFI-newdata/analysis/rfi/Plot06_26_2014_1.pdf}
    \includegraphics[width=\textwidth,height=1.4\textheight,keepaspectratio]{PlotmodelcbcdateGD.png}
    \end{figure}
\end{frame}

\begin{frame}[fragile]
\frametitle{Comparison G8 Data and Single End Count Data G9}

% taken from the file 
When FDR is controlled at $20\%$ for DE Analysis of G8 data, and FDR is controlled at $5\%$ for DE Analysis of single end count G9 data, paired end count G9 data, there are 18 common DE genes between RFI lines
<<>>=
gene_list <- c("ENSSSCG00000009051", "ENSSSCG00000027314", "ENSSSCG00000006457", 
"ENSSSCG00000002792", "ENSSSCG00000007517", "ENSSSCG00000024791", 
"ENSSSCG00000000625", "ENSSSCG00000008614", "ENSSSCG00000012141", 
"ENSSSCG00000016426", "ENSSSCG00000013293", "ENSSSCG00000007656", 
"ENSSSCG00000021646", "ENSSSCG00000008193", "ENSSSCG00000015766", 
"ENSSSCG00000006309", "ENSSSCG00000006378", "ENSSSCG00000006912"
)
gene_list
@
and among those there are 2 genes
with log2fc $\geq 1$. 
<<>>=
commongenel2fc1 <- c("ENSSSCG00000000625", "ENSSSCG00000015766")
commongenel2fc1
@
\end{frame}
\end{document}