\documentclass{article}
% \usepackage[sc]{mathpazo}
% \usepackage[T1]{fontenc}
\usepackage{geometry}
\usepackage{amsmath, amssymb, mathtools }
\usepackage{enumerate}
\usepackage{array}
\usepackage{fancyhdr}
\usepackage{verbatim}
\usepackage{color}
\usepackage{pstricks}
% \usepackage{longtable}
% \usepackage{fancyvrb}
% \usepackage{fancyhdr}
\usepackage{eqnarray}
%\pagestyle{fancy}
\usepackage{psfrag}
\usepackage{epsfig,epsf}
\usepackage{pstricks}
\geometry{verbose,tmargin=2.5cm,bmargin=2.5cm,lmargin=2.5cm,rmargin=2.5cm}
\setcounter{secnumdepth}{2}
\setcounter{tocdepth}{2}
\usepackage{url}
\usepackage[unicode=true,pdfusetitle,
            bookmarks=true,bookmarksnumbered=true,bookmarksopen=true,bookmarksopenlevel=2,
            breaklinks=false,pdfborder={0 0 1},backref=false,colorlinks=false]
{hyperref}
\hypersetup{
  pdfstartview={XYZ null null 1}}
\usepackage{breakurl}
\begin{document}



<<setup, include=FALSE>>=
#render_listings()
#pdf.options(useDingbats = TRUE)
opts_chunk$set(fig.path='figure/beamer-',fig.align='center',fig.show='hold',size='footnotesize',message=FALSE,error=FALSE,warning=FALSE)
opts_chunk$set(fig.width=5, fig.height=5, out.width='.6\\linewidth', fig.align='center')

@

<<echo=FALSE,results='hide'>>=
# some setup
options(width=60) # make the printing fit on the page
set.seed(1121) # make the results repeatable
require(xtable)
require(PASWR)
@

\title{Analyze RNASeq Data from G9P2 RFI Lines Using QuasiSeq Package}

\author{Yet Nguyen}

\maketitle
<<echo=FALSE>>=
options(width=90)
@

\section*{Results of Model 11}
<<>>=

## Covariate Set
covset <- read.csv("covset.csv")
attach(covset)

## Load the result of model 11
load("Model11_result.RData") 

## To get P.values of all Line Testings, use result$P.values[[3]][,"Line]
## To get Q.values of Line Testings, use result$Q.values[[3]][,"Line]

scount <- read.table("single end uniquely mapped reads count table for Yet.txt", 
                     header = T)

## List of Genes used to find DE Genes

counts <- as.matrix(scount[rowSums(scount[,-1]>0)>3&
                           rowMeans(scount[,-1])>1 & 
                             rowSums(scount[,-1][,Line ==1] > 0) >0 &
                             rowSums(scount[,-1][, Line ==2] >0) >0 ,-1])


## List of Line DE Genes when FDR is controled at 0.05, 0.10, 0.15

degene05 <- which(result$Q.values[[3]][,"Line"]<=0.05)
scount[rownames(counts[degene05,]), 1] # List of Line DE Genes when FDR is controled at 0.05
degene10 <- which(result$Q.values[[3]][,"Line"]<=0.10)
degene15 <- which(result$Q.values[[3]][,"Line"]<=0.15)

## Total number of Line  DE Genes when FDR is controlled at 0.05, 0.10, 0.15

degene <- c(length(degene05), length(degene10), length(degene15))

## List of Line DE Genes with log(FC) >=1 when FDR is controlled at 0.05, 0.10, 0.15

lf105 <- degene05[abs(log(apply(counts[degene05, Line ==1]+1, 1, mean)/apply(counts[degene05, Line ==2]+1, 1, mean))) >=1]

scount[rownames(counts[lf105,]), 1] # List of  Line DE Genes with log(FC) >=1 when FDR is controled at 0.05

lf110 <-degene10[abs(log(apply(counts[degene10, Line ==1]+1, 1, mean)/apply(counts[degene10, Line ==2]+1, 1, mean))) >=1]

 scount[rownames(counts[lf110,]), 1] # List of  Line DE Genes with log(FC) >=1 when FDR is controled at 0.10

lf115 <- degene15[abs(log(apply(counts[degene15, Line ==1]+1, 1, mean)/apply(counts[degene15, Line ==2]+1, 1, mean))) >=1]
 scount[rownames(counts[lf115,]), 1]# List of  Line DE Genes with log(FC) >=1 when FDR is controled at 0.15

## Total number of Line DE Genes with log(FC) >=1 when FDR is controlled at 0.05, 0.10, 0.15

lf1 <- c(length(lf105), length(lf110), length(lf115))
@



<<results='asis'>>=
# Summary table
out <- data.frame(FDR =c(0.05, 0.10, 0.15), degene = degene, lf1 = lf1)
colnames(out) <- c("FDR", "DEGs", "log(FC)>=1")
xtable(out)
@

\end{document}
