\documentclass[handout,xcolor=pdftex,dvipsnames,table]{beamer} 

\usecolortheme[RGB={0,0,144}]{structure}
%\usetheme{AnnArbor}\usecolortheme{beaver}
\usetheme{CambridgeUS}\usecolortheme{dolphin}

\usepackage{graphicx}
\usepackage[space]{grffile}
\usepackage{verbatim,xmpmulti,color,multicol,multirow}
\setlength{\unitlength}{\textwidth}  % measure in textwidths
\usepackage[normalem]{ulem}
\usepackage{amssymb,amsmath,latexsym}
\usepackage{booktabs}
\usepackage{array}
\newcolumntype{L}{>{$}l<{$}}
\newcolumntype{C}{>{$}c<{$}}
\newcolumntype{R}{>{$}r<{$}}
\newcommand{\nm}[1]{\textnormal{#1}}


%\usepackage{beamerthemesplit}
\setbeamertemplate{navigation symbols}{}
\setbeamertemplate{caption}[numbered]
%\setbeamercolor{alerted text}{fg=red}
%\setbeamertemplate{block body theorem}{bg=orange}
\setkeys{Gin}{width=0.6\textwidth}


%\SweaveOpts{concordance=TRUE}

<<setup, include=FALSE>>=
#render_listings()
#pdf.options(useDingbats = TRUE)
opts_chunk$set(fig.path='figure/beamer-',fig.align='center',fig.show='hold',size='footnotesize',echo=FALSE,message=FALSE,error=FALSE,warning=FALSE)
opts_chunk$set(fig.width=5, fig.height=5, out.width='.6\\linewidth', fig.align='center')

@

<<echo=FALSE,results='hide'>>=
# some setup
options(width=60) # make the printing fit on the page
set.seed(1121) # make the results repeatable
require(xtable)
require(PASWR)
@

\title[RNASeq Analysis]{Analyze RNASeq Data from G9P2 RFI Lines Using QuasiSeq Package}


% \author[Yet Nguyen, Dan Nettleton]{Yet Nguyen \and Dan Nettleton}
% \institute[ISU]{Iowa State University}

%

\begin{document}
\frame{\maketitle}
\frame {
\frametitle{Table of Contents}
\begin{itemize}
\setlength{\itemsep}{.2in}
\item[1.] Data Summary
\item[2.] Model Selection 
\begin{itemize}
\item Backward Model Selection
\item AIC Model Selection
\end{itemize}
\item[3.] Results
\end{itemize}
}


\section{Data Summary}
\begin{frame}[fragile]
\frametitle{RNASeq Data Summary}
\begin{itemize}
\setlength{\itemsep}{.25in}
\item RNASeq data set is a  25320 $\times$ 31 table of count data  corresponding to 25320 genes of 31  pigs from 2 Lines (high RFI Line and low RFI Line) and 2 Diets (high energy Diet 1 and low energy Diet 2).

\item  For Diet 1, the RNA data are from 7 low RFI line pigs and 8  high RFI line pigs. 
For Diet 2, the RNA data are from 8 low RFI line pigs and 8  high RFI line pigs.

\item The number of genes analyzed is 15058. Those are genes with average counts greater than one and for which there are at least four samples with non-zero counts.  
\end{itemize}
 <<echo=FALSE,include=FALSE>>=
dir.source <- "/run/user/1000/gvfs/smb-share:server=cyfiles.iastate.edu,share=09/22/ntyet/stevescode/QuasiSeq_1.0-2/QuasiSeq/R/"
source(paste(dir.source, "QL.fit.R",sep=""))
source(paste(dir.source, "NBDev.R",sep =""))
source(paste(dir.source, "PoisDev.R",sep =""))
source(paste(dir.source, "QL.results.R",sep =""))

# # ### Reading data #######
# # scount <- read.table("/home/ntyet/research/RFI-newdata/Data for Yet/single end uniquely mapped reads count table for Yet.txt", 
# #                      header = T)
# # cbc <- read.table('/home/ntyet/research/RFI-newdata/Data for Yet/CBC data for pigs with RNA-seq data avaible.txt',
# #                   header =T)
# # 
# # metadata <- read.table("/home/ntyet/research/RFI-newdata/Data for Yet/meta_data_RNA-seq_G9P2.txt", 
# #                        header = T)
# # 
# # rfiadj <- read.csv("/home/ntyet/research/RFI-newdata/Data for Yet/g8p2_g9p2-rfiadj-FINAL_Jan_2014_rfiadjusted.csv", 
# #                      header = T)
# 
# 
### Reading data #######
scount <- read.table("single end uniquely mapped reads count table for Yet.txt", 
                     header = T)
cbc <- read.table('CBC data for pigs with RNA-seq data avaible.txt',
                  header =T)

metadata <- read.table("meta_data_RNA-seq_G9P2.txt", 
                       header = T)

rfiadj <- read.csv("g8p2_g9p2-rfiadj-FINAL_Jan_2014_rfiadjusted.csv", 
                     header = T)
# resultdir <- "/run/user/1000/gvfs/smb-share:server=cyfiles.iastate.edu,share=09/22/ntyet/R/RA/Data/RFI-newdata/result"
# #resultdir <- "U:/R/RA/Data/RFI-newdata/result"
# # scount <- read.table("single end uniquely mapped reads count table for Yet.txt", 
# #                      header = T)
# # cbc <- read.table(paste(resultdir, "/rfi/CBC data for pigs with RNA-seq data avaible.txt", sep = ""),
# #                   header =T)
# # 
# # metadata <- read.table("meta_data_RNA-seq_G9P2.txt", 
# #                        header = T)
# # 
# # rfiadj <- read.csv("g8p2_g9p2-rfiadj-FINAL_Jan_2014_rfiadjusted.csv", 
# #                    header = T)
# 
##### cleaning data####

cbc <- cbc[order(cbc$ear), ]
metadata <- metadata[order(metadata$idpig), ]
rfiadj <- rfiadj[order(rfiadj$idpig),]

fullidpig <- as.numeric(paste("20900", metadata$idpig, sep = ""))

covset <- cbind(metadata[, -4], rfiadj[rfiadj$idpig %in% fullidpig, c("rfi.ADJUSTED")], 
                cbc[, c("iddam", "idsire", "Neutrophil", "Lymphocyte", "Monocyte",
                        "Eosinophil", "Basophil" )])
colnames(covset) <- c("idpig", "Line", "Diet",  "Block", "Blockorder", "Concb", 
                      "RINb", "Conca", "RINa", "RFI",
                      "iddam", "idsire", "neut",   
                      "lymp", "mono","eosi", "baso")

#####set of covariates considered ####
covset <- cbind(covset, lneut = log(covset$neut), llymp = log(covset$lymp), 
                lmono = log(covset$mono), leosi = log(covset$eosi), 
                lbaso = log(covset$baso)) 
covset$Line <- as.factor(covset$Line)
covset$Diet <- as.factor(covset$Diet)
covset$Block <- as.factor(covset$Block)
covset$Blockorder <- as.factor(covset$Blockorder)
covset$iddam <- as.factor(covset$iddam)
covset$idsire <- as.factor(covset$idsire)
levels(covset$idsire) <- 1:11
levels(covset$iddam) <- 1:20

#covset[, c("iddam", "idsire")]
# #detach(covset)
 attach(covset)
counts <- as.matrix(scount[rowSums(scount[,-1]>0)>3&
                           rowMeans(scount[,-1])>1 & 
                             rowSums(scount[,-1][,Line ==1] > 0) >0 &
                             rowSums(scount[,-1][, Line ==2] >0) >0 ,-1])


# dim(scount)
# dim(counts)
@

\end{frame}




%%%%%%%%%%%%%%%%
\frame{
\frametitle{Metadata Summary}
\begin{itemize}
\setlength{\itemsep}{.25in}
\item The available metadata consists of infomation of  9 covariates for 31 samples of 31 pigs.
\item Factors: Diet (2 levels), Line (2 levels), Block (4 levels), Blockorder (8 levels).
\item Quantitative covariates: RFI (RFI values) , RINb (RIN before globin depletion), RINa (RIN after globin depletion), Concb (RNA Concentration before globin depletion), Conca(RNA Concentration after globin depletion).
\end{itemize}
}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

% \begin{frame}
% \frametitle{Goals of the Analysis}
% \begin{itemize}
% \setlength{\itemsep}{.25in}
% \item Identify genes that are differentially expressed between Diets, Lines and/or show Diet$\times$Line  interaction.
% \item Identify genes associated with RFI values.
% \item Account for effects of other covariates and factors on the RNA data.
% \end{itemize}
% \end{frame}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%
% 
% \begin{frame}
% \frametitle{Analyze RNA Data Using Quasi-Likelihood Method}
% \begin{itemize}
% \setlength{\itemsep}{.25in}
% \item To accomplish these goals, we use Quasi-likelihood with shrunken dispersion estimates proposed by S. Lund et al.\ (\emph{Statistical Applications in Genetics and Molecular Biology}, 2012).
% 
% \item Considering various design matrices that include different sets of covariates, we obtain a histogram of $p$-values for each factor and covariate of each model.
% 
% % \item Compare $p$-values of Diet, Line, RFI.value for those models.
% % 
% % \item Compute mean and median of estimated dispersion parameters, and AIC for each model, to choose the "best" model among examined models.
% \end{itemize}
% \end{frame}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section{Model Selection}
\begin{frame}[fragile]
\frametitle{Backward Model Selection}
\begin{figure}[h!]
    \centering
    \includegraphics[width=\textwidth,height=0.8\textheight,keepaspectratio]{/home/ntyet/research/RFI-newdata/analysis/rfi/Plot06_26_2014_1.pdf}
    \end{figure}
\end{frame}


\begin{frame}[fragile]
\frametitle{AIC Model Selection}
\begin{itemize}
\item \emph{AIC:} Obtain Log Maximum Likelihood and calculate AIC for each gene,
\[
AIC = -2 Log.Max.Lik + 2 p,
\]
where $p$ is the number of parameters in the model. 
Then, calculate mean of those values across all genes for each model.

\item AIC of above models
<<>>=
aic <- read.table("model_aic.txt")[-1,]

names(aic) <- paste("M", 2:11, sep = "")
aic
@
\item The best model is Model 11, which is consistent with the Backward Selection method.
\end{itemize}
\end{frame}
\section{Results}
\begin{frame}[fragile]
\frametitle{Results of Model 11}
\begin{itemize}
\item 
Estimated number of DE Genes between two RFI Lines
<<>>=
load("Model11_fit.RData")
load("Model11_result.RData")
dim(result$Q.values[[3]])[1]- result$m0[3,1]
@

\item When FDR is controlled at 0.05, 0.10, 0.15, the number of DE Genes between two RFI Lines (DEGs) and the number of DE Genes between two RFI Lines with log(fold change) at least 1 (log(FC) $>1$) are shown in the table below
<<results ='asis'>>=
degene <- c(sum(result$Q.values[[3]][,"Line"]<=0.05),
sum(result$Q.values[[3]][,"Line"]<=0.1),
sum(result$Q.values[[3]][,"Line"]<=0.15))

degene05 <- which(result$Q.values[[3]][,"Line"]<=0.05)
degene10 <- which(result$Q.values[[3]][,"Line"]<=0.10)
degene15 <- which(result$Q.values[[3]][,"Line"]<=0.15)

lf1 <- c(sum(abs(log(apply(counts[degene05, Line ==1]+1, 1, mean)/apply(counts[degene05, Line ==2]+1, 1, mean))) >=1),
sum(abs(log(apply(counts[degene10, Line ==1]+1, 1, mean)/apply(counts[degene10, Line ==2]+1, 1, mean))) >=1),
sum(abs(log(apply(counts[degene15, Line ==1]+1, 1, mean)/apply(counts[degene15, Line ==2]+1, 1, mean))) >=1))

out <- data.frame(FDR =c(0.05, 0.10, 0.15), degene = degene, lf1 = lf1)
colnames(out) <- c("FDR", "DEGs", "log(FC)>1")
xtable(out)
@
\end{itemize}
\end{frame}

\end{document}